### PR TITLE
Add model of karmada

### PR DESCRIPTION
**Pull request by liangyuanpeng**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

Add model of karmada, and provide crds of policy first.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: karmada/0.1.0/artifacthub-pkg.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  ```
+  kpm pull karmada:0.1.0
+  ```
+maintainers:
+- name: kcl-lang.io
+  email: kcl-lang.io@domainsbyproxy.com
+- name: Lan Liang
+  email: gcslyp@gmail.com
+provider:
+  name: kcl-lang.io
```
Comment (liangyuanpeng) [inline karmada/0.1.0/artifacthub-pkg.yaml:27]: Should i put kcl-lang.io to it?

File: karmada/kcl.mod [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[package]
+name = "karmada"
+edition = "0.1.0"
+version = "0.1.0"
+
+[dependencies]
+k8s = "1.27"
```
Comment (liangyuanpeng) [inline karmada/kcl.mod:7]: This model is generate with kcl-openapi, how can i get this k8s version? And i think i should be not push the files of `k8s/apimachinery/xxx`.

**Review by liangyuanpeng — COMMENTED** [review]
Actually, i have some question at comment for add a new model.

And i have not any test for this model.

Remind me if i missed something,Thanks.

File: karmada/kcl.mod (hunk 2 above)
Comment (Peefy) [reply karmada/kcl.mod:7]: > This model is generate with kcl-openapi, how can i get this k8s version? And i think i should be not push the files of `k8s/apimachinery/xxx`.

Yes, U can use the `kpm` tool to add this dependecy e.g., `kpm add k8s:1.27`. And U can remove the k8s folder.

**Comment by Peefy** [comment]
> Actually, i have some question at comment for add a new model.
> 
> And i have not see the any test for models of this repo.
> 
> Remind me if i missed something,Thanks.

We are currently refactoring the testing tool, which is expected to be available in the official version of v0.7.0. At that time, we will improve the process of writing test cases and test cases (expected to write unit tests like Go).

File: karmada/kcl.mod (hunk 2 above)
Comment (Peefy) [inline karmada/kcl.mod:7]: U can use `kcl mod add k8s:1.27` with the kcl v0.7.0 version to update the kcl.mod.lock file.

File: karmada/0.1.0/artifacthub-pkg.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+version: 0.1.0
```
Comment (Peefy) [inline karmada/0.1.0/artifacthub-pkg.yaml:1]: U can move this file into the `karmada` folder. The `karmada/0.1.0/artifacthub-pkg.yaml` file will be generated automatically through CI.

**Review by Peefy — COMMENTED** [review]
Thank you for your contribution ❤️. I have left some comments.

**Pull request merged**